scene/gui/button.cpp

```diff
@@ -101,7 +102,8 @@ void Button::_notification(int p_what) {
 			case DRAW_HOVER: {
 
 				style = get_stylebox("hover");
-				style->draw(ci, Rect2(Point2(0, 0), size));
+				if (!flat)
+					style->draw(ci, Rect2(Point2(0, 0), size));
```

Comment:
@volzhs i think this should be the correct behavior (flatness should be an general property), otherwise the flatness is kinda pointless. docs description should be changed.